lib/sensu/utilities.rb

```diff
@@ -51,6 +51,31 @@ def deep_merge(hash_one, hash_two)
       merged
     end
 
+    # Creates a deep dup of basic ruby objects with support for walking
+    # hashes and arrays.
+    #
+    # @param obj [Object]
+    # @return [obj] a dup of the original object.
+    def deep_dup(obj)
+      if obj.class == Hash
+        new_obj = obj.dup
+        new_obj.each do |key, value|
+          new_obj[deep_dup(key)] = deep_dup(value)
```

Comment:
There's a very good chance key is usually (or always) a symbol, therefor, this will return the original object.